volatility/framework/plugins/mac/kevents.py

```diff
@@ -0,0 +1,185 @@
+    """ Lists event handlers registered by processes """ 
+
+    event_types = { 1 : "EVFILT_READ", 
+                    2 : "EVFILT_WRITE", 
+                    3 : "EVFILT_AIO", 
+                    4 : "EVFILT_VNODE", 
+                    5 : "EVFILT_PROC", 
+                    6 : "EVFILT_SIGNAL",
+                    7 : "EVFILT_TIMER", 
+                    8 : "EVFILT_MACHPORT", 
+                    9 : "EVFILT_FS", 
+                    10 : "EVFILT_USER", 
+                    12 : "EVFILT_VM" }
+
+    vnode_filters = [("NOTE_DELETE", 1), 
+                     ("NOTE_WRITE",  2), 
+                     ("NOTE_EXTEND", 4), 
+                     ("NOTE_ATTRIB", 8),
+                     ("NOTE_LINK",   0x10), 
+                     ("NOTE_RENAME", 0x20), 
+                     ("NOTE_REVOKE", 0x40)]
+
+    proc_filters = [("NOTE_EXIT",       0x80000000), 
+                    ("NOTE_EXITSTATUS", 0x04000000), 
+                    ("NOTE_FORK",       0x40000000),
+                    ("NOTE_EXEC",       0x20000000), 
+                    ("NOTE_SIGNAL",     0x08000000), 
+                    ("NOTE_REAP",       0x10000000)]
+
+    timer_filters = [("NOTE_SECONDS",   1), 
+                     ("NOTE_USECONDS", 2), 
+                     ("NOTE_NSECONDS", 4), 
+                     ("NOTE_ABSOLUTE", 8)]
+
+    all_filters = {4 : vnode_filters, # EVFILT_VNODE
+                   5 : proc_filters,  # EVFILT_PROC
+                   7 : timer_filters  # EVFILT_TIMER
+                   }
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "darwin", description = "Mac kernel"),
+            requirements.PluginRequirement(name = 'pslist', plugin = pslist.PsList, version = (2, 0, 0)),
+            requirements.VersionRequirement(name = 'macutils', component = mac.MacUtilities, version = (1, 2, 0)),
+            requirements.ListRequirement(name = 'pid',
+                                         description = 'Filter on specific process IDs',
+                                         element_type = int,
+                                         optional = True)
+        ]
+
+    def _parse_flags(self, filter_index, filter_flags):
+        if filter_flags == 0 or filter_index not in self.all_filters:
+            return ""
+        
+        context = []
+
+        filters = self.all_filters[filter_index]
+        for flag, index in filters:
+            if filter_flags & index == index:
+                context.append(flag)
+
+        return ",".join(context)
+
+    @classmethod
+    def _walk_klist_array(cls, kernel, fdp, array_pointer_member, array_size_member):
```

Comment:
Looks good, just want to check that these aren't ever intended to be used separately (since they're private)?